### PR TITLE
Work on DCurves

### PR DESCRIPTION
**Pull request by mateuszbaran**

### CODE REVIEW

File: src/DCurves.jl [hunk 1]
```diff
@@ -0,0 +1,68 @@
+
+"""
+    DCurves(M::Manifold, grid::AbstractVector)
+
+Space of curves on manifold `M` discretized on the given `grid`.
+"""
+struct DCurves{𝔽,TM<:Manifold{𝔽},TG<:AbstractVector} <:
+       Manifolds.AbstractPowerManifold{𝔽,TM,Manifolds.ArrayPowerRepresentation}
+    manifold::TM
+    grid::TG
```
Comment (kellertuer) [inline src/DCurves.jl]: is this a vector of points, i.e. an irregular grid?

File: src/FunManifolds.jl [hunk 2]
```diff
@@ -91,23 +94,25 @@ end
...
-        atoldefault(M.M, x1, x2)
+        atoldefault(M, x1, x2)
     else
         abstol
     end
 
     return (rtol, atol)
 end
 
+include("DCurves.jl")
```
Comment (kellertuer) [inline src/FunManifolds.jl]: Maybe `DiscretizedCurves` is a better name?

**Review by kellertuer — COMMENTED** [review]
I don't understand this fully, maybe a little more documentation would help, but the approach looks nice. Isn't that also a manifold, the one of discretised curves?

**Comment by mateuszbaran** [comment]
It's very much a work-in-progress, I'm porting my old code and it's not even fully working. I'll let you know once it gets somewhat OK.

File: src/DCurves.jl (hunk 1 above)
Comment (mateuszbaran) [reply src/DCurves.jl]: So far I've only used uniform grids but here I allow irregular grid as well. It's supposed to be a vector of real numbers where the curve was discretized.

File: src/FunManifolds.jl (hunk 2 above)
Comment (mateuszbaran) [reply src/FunManifolds.jl]: Yes, I'll most likely change it at a later stage of porting.

**Comment by kellertuer** [comment]
Yes, nevermind, I just noticed those two points :) Feel free to mark this, if its ready for review.

File: src/DCurves.jl (hunk 1 above)
Comment (kellertuer) [reply src/DCurves.jl]: I was thinking of irregular ones, since for regular ones just a set of stepsizes is necessary.

**Comment by mateuszbaran** [comment]
This PR is in a state that is anyway better than current master branch so I'll just merge it and do new PRs with further improvements.

**Pull request merged**